subprojects/file-watching/src/test/groovy/org/gradle/internal/watch/registry/impl/AbstractFileWatcherUpdaterTest.groovy

```diff
@@ -225,6 +227,55 @@ abstract class AbstractFileWatcherUpdaterTest extends Specification {
         vfsHasSnapshotsAt(newestRegisteredWatchableHierarchy)
     }
 
+    def "does not start watching unsupported file system for default watch mode"() {
+        def unsupportedFileSystemMountPoint = file("unsupported").createDir()
+        def unwatchableContent = unsupportedFileSystemMountPoint.file("file.txt").createFile()
+        def unsupportedFileSystem = Stub(FileSystemInfo) {
+            getMountPoint() >> unsupportedFileSystemMountPoint
+            getFileSystemType() >> "unsupported"
+        }
+        watchableFileSystemDetector.detectUnsupportedFileSystems() >> { it -> Stream.of(unsupportedFileSystem) }
+
+        when:
+        buildStarted(WatchMode.DEFAULT)
+        registerWatchableHierarchies([unsupportedFileSystemMountPoint])
+        addSnapshot(snapshotRegularFile(unwatchableContent))
+        then:
+        vfsHasSnapshotsAt(unwatchableContent)
+        0 * _
+
+        when:
+        buildFinished(Integer.MAX_VALUE, WatchMode.DEFAULT)
+        then:
+        !vfsHasSnapshotsAt(unwatchableContent)
+        0 * _
+    }
+
+    def "does start watching unsupported file system when watching is enabled"() {
```

Comment:
Shall we use the terminology "force enabled" here? Or maybe "enforced"? I think just saying "enabled" is a bit confusing. (And this applies to documentation, too.)

Reply:
We could do something like this. Though I'd prefer doing this as part of a different PR.